lib/osf-components/addon/components/validated-model-form/component.ts

```diff
@@ -72,19 +80,34 @@ export default class ValidatedModelForm<M extends ValidatedModelName> extends Co
         super(...args);
 
         assert('Can only pass either a model or a modelName', !(Boolean(this.model) && Boolean(this.modelName)));
-
         if (!this.model && this.modelName) {
             this.model = this.store.createRecord(this.modelName, this.modelProperties);
         }
+        if (!this.changeset && this.model) {
+            set(this, 'changeset', this.buildChangeset(this.model));
+        }
     }
 
     willDestroy() {
-        if (this.model) {
-            if (this.onWillDestroy !== undefined) {
-                this.onWillDestroy(this.model);
-            } else if (!this.saved) {
-                this.model.unloadRecord();
-            }
+        if (this.changeset) {
+            this.changeset.rollback();
         }
     }
+
+    // Lifted wholesale from https://github.com/offirgolan/ember-changeset-cp-validations/blob/master/addon/index.js
```

Comment:
Would it make sense to install the addon and
```ts
import { buildChangeset } from 'ember-changeset-cp-validations'
```
instead of lifting wholesale?

Reply:
Not really, because the project is not frequently updated, and is out of date. So for the 14 lines we need from it, I'm okay with maintaining this one function rather than adding the dependency.